internal/kafka/internal/workers/kafka_mgrs/promotion/promotion_kafkas_mgr_test.go

```diff
@@ -0,0 +1,325 @@
+						return errors.NewServiceErrorBuilder().Recoverable().WithReason("Test delete failure").Build()
+					},
+				},
+			},
+		},
+		{
+			name:         "Test fail reserving quota",
+			wantErrCount: 1,
+			expect: expect{
+				wantDesiredBillingModel: "standard",
+				wantActualBillingModel:  "standard",
+				kafkaService_ListKafkasToBePromoted: KafkaService_ListKafkasToBePromotedExpect{
+					calls: 1,
+				},
+				kafkaService_Update: KafkaService_UpdateExpect{
+					promotionStatus: dbapi.KafkaPromotionStatusFailed,
+					calls:           1,
+				},
+				quotaService_ReserveQuotaIfNotAlreadyReserved: QuotaService_ReserveQuotaIfNotAlreadyReservedExpect{
+					calls: 1,
+				},
+			},
+			fields: fields{
+				kafkaConfig: config.KafkaConfig{
+					Quota: &config.KafkaQuotaConfig{
+						Type:                         api.AMSQuotaType.String(),
+						AllowDeveloperInstance:       true,
+						MaxAllowedDeveloperInstances: 10,
+					},
+					SupportedInstanceTypes: &config.KafkaSupportedInstanceTypesConfig{
+						Configuration: config.SupportedKafkaInstanceTypesConfig{
+							SupportedKafkaInstanceTypes: []config.KafkaInstanceType{
+								{
+									Id:          "standard",
+									DisplayName: "standard",
+									Sizes:       nil,
+									SupportedBillingModels: []config.KafkaBillingModel{
+										{
+											ID:               "eval",
+											AMSResource:      "rhosak",
+											AMSProduct:       "RHOSAKEval",
+											AMSBillingModels: []string{"standard"},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				kafkaService: &services.KafkaServiceMock{
+					ListKafkasToBePromotedFunc: func() ([]*dbapi.KafkaRequest, *errors.ServiceError) {
+						return []*dbapi.KafkaRequest{
+							{
+								Meta: api.Meta{
+									ID: "123-kafka",
+								},
+								ActualKafkaBillingModel:  "eval",
+								DesiredKafkaBillingModel: "standard",
+								InstanceType:             types.STANDARD.String(),
+							},
+						}, nil
+					},
+					UpdateFunc: func(kafkaRequest *dbapi.KafkaRequest) *errors.ServiceError {
+						return nil
+					},
+				},
+				quotaService: &services.QuotaServiceMock{
+					ReserveQuotaIfNotAlreadyReservedFunc: func(kafka *dbapi.KafkaRequest) (string, *errors.ServiceError) {
+						return "", errors.NewServiceErrorBuilder().WithReason("Test reserve failure").Build()
+					},
+				},
+			},
+		},
+	}
+
+	for _, testcase := range tests {
+		tt := testcase
+
+		t.Run(tt.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
+			k := NewPromotionKafkaManager(
+				workers.Reconciler{},
+				tt.fields.kafkaService,
+				&tt.fields.kafkaConfig,
+				&services.QuotaServiceFactoryMock{
+					GetQuotaServiceFunc: func(quotaType api.QuotaType) (services.QuotaService, *errors.ServiceError) {
+						return tt.fields.quotaService, nil
+					},
+				},
+			)
+
+			errs := k.Reconcile()
+			g.Expect(errs).To(gomega.HaveLen(tt.wantErrCount))
+
+			//if len(errs) == 0 {
```

Comment:
This comment does not seem accurate.
If len(errs) == 0 the following code would still be executed, assuming that the gomega expectation executed above matches